File: src/main/java/ru/isb/bot/utils/JsonUtils.java
package ru.isb.bot.utils;

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.SneakyThrows;

public class JsonUtils {

    private static final ObjectMapper mapper = new ObjectMapper();

    static {
        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
    }

    @SneakyThrows
    public static <T> T parseStringJsonToObject(String content, Class<T> clazz) {
        return mapper.readValue(content, clazz);
    }

    @SneakyThrows
    public static String parseObjectToString(Object object) {
        return mapper.writeValueAsString(object);
    }
}

File: src/main/java/ru/isb/bot/utils/JsonUtils.kt
package ru.isb.bot.utils

import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.ObjectMapper
import lombok.SneakyThrows

class JsonUtils {

    companion object {
        private val mapper = ObjectMapper()

        init {
            mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
        }

        @SneakyThrows
        fun <T> parseStringJsonToObject(content: String?, clazz: Class<T>?): T? {
            return mapper.readValue(content, clazz)
        }

        @SneakyThrows
        fun parseObjectToString(`object`: Any?): String {
            return mapper.writeValueAsString(`object`)
        }
    }
}